Entities/Characters/Builder/BuilderAnim.as

```diff
@@ -165,16 +177,26 @@ void onTick(CSprite@ this)
 							hitting_stone = true;
 						}
 
+						if (attacked_name == "bridge" ||
+							attacked_name == "wooden_platform" ||
+							attacked.hasTag("door") ||
+							attacked_name == "ladder" ||
+							attacked_name == "spikes"
+							)
+						{
+							hitting_built = true;
+						}
+
 					}
 				}
 
 				if (hitting_wood)
 				{
-					attack_anim = "chop";
+					attack_anim = hitting_built ? "chop_fast" : "chop";
 				}
 				else if(hitting_stone)
 				{
-					attack_anim = "strike";
+					attack_anim = hitting_built ? "strike_fast" : "strike";
 				}
```

Comment:
You can remove a lot of code in BuilderAnim.as & builder.cfg if you add an `if (hitting_built)` with smart usage of `AddFrame`/`RemoveFrame`/`getFrame`/`getFramesCount`

Reply:
resolved if i understood properly